Backpack-SwiftUI/Skeleton/Classes/BPKShimmer.swift

```diff
@@ -19,49 +19,34 @@
 import SwiftUI
 
 struct BPKShimmer: ViewModifier {
-    @State private var phase: CGFloat = 0
+    @State private var offset = 0.0
     private let duration = 1.0
     private let bounce = false
-    private let delay = 0.2
-
+    private let bandSize = 0.3
+    
     public func body(content: Content) -> some View {
         content
-            .modifier(ShimmerMask(phase: phase).animation(
-                Animation.linear(duration: duration)
-                    .repeatForever(autoreverses: bounce)
-            ))
-            .onAppear { phase = 0.8 }
+            .mask(linearGradient)
+            .animation(linearAnimation, value: offset)
+            .onAppear {
+                offset = 1.0 + bandSize
+            }
     }
-
-    struct ShimmerMask: AnimatableModifier {
```

Comment:
`AnimatableModifier` is deprecated: https://developer.apple.com/documentation/swiftui/animatablemodifier

Reply:
Nice one!